### ISSUE TITLE
Regex matching for command line version of RTK not working

### ISSUE THREAD

**Issue by ZijunHui** [RTKConsortium/RTK#285]
### Description

I have compiled RTK with Insight toolkit 5.0.1. I am using the command line compiled version of RTK.

For any command line program that has the '-r' argument (e.g. rtkfdk), specifying a regex that does match the intended files causes them to be ignored. Assume I have a folder "proj" with 20 files ending in '.hnc'. I use the following command.

```
rtkfdk -o recons\\fFDK.mha -p proj -r hnc -g Geometry.xml -v on --dimension 100,100 --spacing 3
```

Using the regex 'hnc' will cause the output to be the following. 

```
Regular expression matches 20 file(s)... 
Ignoring file: proj/proj1.hnc,
Ignoring file: proj/proj2.hnc,
Ignoring file: proj/proj3.hnc,
...
```

I assume the regex was correctly matched, since all files are matched, yet somehow the files are igored? This also results in the following error being immediately thrown

```
ExceptionObject caught with reader->Update() in file C:\...\InsightToolkit-5.0.1\Modules\Remote\RTK\applications\rtkfdk\rtkfdk.cxx line 63 
 
itk::ExceptionObject (000000AECA8FF760) 
Location: "unknown"  
File: C:\Users\allen\Desktop\InsightToolkit-5.0.1\Modules\Core\Common\src\itkProcessObject.cxx 
Line: 1403 
Description: itk::ERROR: StreamingImageFilter(000001FE9F936D30): Input Primary is required but not set.
```

Also to add, an unknown version of the same program works fine and runs without errors.

### Possible cause:

From reading the source code, I believe, the issue is caused the following lines. It assumes the images are readable by ITK. However, reading the [wiki](https://github.com/SimonRit/RTK/wiki/File-formats), it appears RTK supports files that ITK does not support, and the from the commit history it appears the hnc reader is written in-house.

https://github.com/SimonRit/RTK/blob/2e8f029d4a9031a24c284d97b07f4cf833c2c094/include/rtkGgoFunctions.h#L151-L178

**Comment by SimonRit** [comment]
RTK currently does not support .hnc. See discussion starting [here](https://public.kitware.com/pipermail/rtk-users/2019-November/010821.html) and #268. It's on my todo list to finalize #268.

**Comment by SimonRit** [comment]
Fixed by #304
